src/LieTheory/WeylGroup.jl

```diff
@@ -1085,3 +1085,10 @@ julia> coefficients(w * x)
 
   return G, MapFromFunc(W, G, iso)
 end
+
+function torsion_subgroup(W::WeylGroup)
+  @req is_finite(W) "An infinite Weyl group does not admit a torsion subgroup"
+  return (W, id_hom(W))
+end
+
+id_hom(W::WeylGroup) = AbstractAlgebra.identity_map(W)
```

Comment:
This is not an actual homomorphism, so I would prefer not to put this into `src` but instead to `experimental/LieTheory`, where we also have a bunch of other functions that way for proper homs of Weyl groups